File: Data Structures & Algorithms/two-integer-sum/submission-0.js
class Solution {
  /**
   * @param {number[]} nums
   * @param {number} target
   * @return {number[]}
   */
  twoSum(nums, target) {
 const temp = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const compliment = target - num;
    if (compliment in temp) {
      return [i, temp[compliment]];
    }
    temp[num] = i;
    }
  }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        int seen[] = new int[nums.length - 1];
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            if (seen[num - 1] == 1) {
                return num;
            } else {
                seen[num - 1] = 1;
            }
        }

        return -1;
    }
}
